Source/monster.cpp

```diff
@@ -1116,7 +1116,7 @@ void NewMonsterAnim(int i, AnimStruct *anim, int md)
 	MonsterStruct *Monst = monster + i;
 	Monst->_mAnimData = anim->Data[md];
 	Monst->_mAnimLen = anim->Frames;
-	Monst->_mAnimCnt = 0;
+	Monst->_mAnimCnt = -2;

```

Comment:
We need to use `-2` instead of `-1` due to an off-by-one mistake in `monsters.cpp` inside `ProcessPlayers`:

https://github.com/diasurgical/devilutionX/blob/4a4486d6161028e69702bedec6cb071e6a5354c3/Source/monster.cpp#L4618-L4634

Notice the check for animation length here:

https://github.com/diasurgical/devilutionX/blob/4a4486d6161028e69702bedec6cb071e6a5354c3/Source/monster.cpp#L4620

Instead of testing with `>` they use `>=`. If we start with `mAnimCnt` as -1, it will be incremented to 0 right away and satisfy this check right on the first game frame, defeating the purpose of the change.

A value of -2 guarantees that it will only increment a frame on the next tick.